### PR TITLE
Disable certain beatmap option buttons when there are no beatmaps selected

### PR DESCRIPTION
**Pull request by nanashi-1**
This PR disables the "Delete all difficulties", "Clear local scores", and "Edit beatmap" options in the `PlaySongSelect` when there is no beatmap selected in the `BeatmapCarousel`.

**Before:**
Screenshot from 2022-11-15 21-07-00
**After:**
Screenshot from 2022-11-15 21-10-08

**Before; when "Delete all difficulties" is clicked, and there is no selected beatmap, this dialog appears:**
Screenshot from 2022-11-15 21-07-07

**Before; when "Edit beatmap" is clicked and quit without saving, this happens:**
Screenshot from 2022-11-15 21-07-27

### CODE REVIEW

File: osu.Game.Tests/Visual/SongSelect/TestScenePlaySongSelect.cs
Comment (frenzibyte) [inline osu.Game.Tests/Visual/SongSelect/TestScenePlaySongSelect.cs:1068]: The null-forgiving operator reads badly especially when used for boolean expressions. We generally use `AsNonNull` instead.

File: osu.Game/Screens/Select/Options/BeatmapOptionsButton.cs
Comment (frenzibyte) [inline osu.Game/Screens/Select/Options/BeatmapOptionsButton.cs:40]: This class already inherits `OsuClickableContainer`, which provides an `Enabled` property. Please use that instead of inventing another field that's opposite to it.

File: osu.Game/Screens/Select/SongSelect.cs
Comment (frenzibyte) [inline osu.Game/Screens/Select/SongSelect.cs:665]: This shouldn't exist. Simplest way to avoid this would be to define a bindable that's updated similar to this method, but is bound directly to all buttons that require a beatmap.

File: osu.Game.Tests/Visual/SongSelect/TestScenePlaySongSelect.cs
Comment (frenzibyte) [inline osu.Game.Tests/Visual/SongSelect/TestScenePlaySongSelect.cs:1169]: Use `ChildrenOfType` rather than exposing every single button.

**Review by frenzibyte — CHANGES_REQUESTED** [review]

**Comment by frenzibyte** [comment]
I've rewritten this PR and moved the state change to happen on the general "options" button instead, as it is supposed to be containing operations that are to be performed on the selected beatmap, making it pointless when none is selected. Looks fine, but will wait for a second approval since I've overhauled this.

**Review by frenzibyte — DISMISSED** [review]

**Comment by peppy** [comment]
Test failures looking very relevant.

**Review by peppy — APPROVED** [review]

**Pull request merged**
